pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -365,6 +415,39 @@ func (c *Controller) Cleanup() error {
 	return nil
 }
 
+// PodEvent reports an event back to a Pod. Can be used with 'kubectl describe pod' or 'kubectl get events'.
+// Important events associated with the Envoy connection should be reported.
+// The method signatures avoid deps on K8S API, so the events can be reported via other mechanisms ( XDS, pubsub)
+func (c *Controller) PodEvent(ns, pod, reason, msg string, warn bool) {
+	if c.Recoder == nil {
+		return
+	}
+	// Event type can be normal and warning
+	evType := v1.EventTypeNormal
+	if warn {
+		evType = v1.EventTypeWarning
+	}
+	// reference is required
+	p, err := c.pods.podInformer.Lister().Pods(ns).Get(pod)
+	var ref runtime.Object
+	if err != nil {
+		log.Infoa("Pod not found, dummy object", err)
```

Comment:
Does the dummy work? If so, why not just always use it?

Reply:
We want the event associated with the real pod. 
I would rather remove the 'dummy' - I don't remember what happens. I'll clean it up.